synapseclient/table.py

```diff
@@ -777,6 +821,51 @@ def __init__(self, values, rowId=None, versionNumber=None, etag=None):
             self.etag = etag
 
 
+class PartialRow(DictObject):
+    """
+    A partial update for a single row.
+    Note "values" is a :py:class:`dict` where:
+        - key is the columnId to the desired row
+        - value is the new desired value for that column
+    The easiest way to know which columnIds to use is to check the `headers` of your most recent query:"
+        query_result = syn.TableQuery("SELECT * FROM syn123")
+        name_to_col_id = {col.name:col.id for col in query_results.headers}
+
+        row_id = #you to set this yourself#
+
+        values_to_change = {'My Row Name': "new value",
+                            'My Other Row': 42}
+        partial_row = PartialRow(values_to_change, row_id, nameToColumnId=name_to_col_id)
+
+    :param values: A dict where:
+                - key is the columnId to the desired row
+                - value is the new desired value for that column
+    :param rowId: THe id of the row to be updated
+    :param etag: used for updating File/Project Views. Not necessary if updating data Tables
+    :param nameToColumnId: Optional map column names to column Ids. If this is provided, the keys of your `values`
+                           dict will be replaced with the column ids in the `nameToColumnId` dict
+
+    """
+    def __init__(self, values, rowId, etag=None, nameToColumnId = None):
+        super(PartialRow, self).__init__()
+        if not isinstance(values, dict):
+            raise ValueError("values must be a dict")
+
+        if isinstance(rowId, (six.integer_types,six.string_types)):
+            try:
+                rowId = int(rowId)
+            except ValueError:
+                raise ValueError("rowId must be an integer or string that represents an integer")
```

Comment:
This seems completely unecessary.  Just do int(rowID)  this will throw a value error if it is not an int.